### PR TITLE
[None][feat] AutoDeploy: Onboard google/gemma-4-31B-it dense model, including nvfp4

### PR DESCRIPTION
**Pull request by suyoggupta**
1. Onboard google/gemma-4-31B-it dense model:
   - Add dense-variant unit tests (decoder layer, full model,
     ConditionalGeneration wrapper, torch.export) to test_gemma4_modeling.py
   - Create gemma4_dense.yaml registry config (triton_paged, world_size=2)
   - Add google/gemma-4-31B-it entry to models.yaml

2. Fix cudaErrorMisalignedAddress in Triton paged attention for head_dim>256:
   - Force SDPA (cuDNN) path for layers with head_dim>256 by bypassing
     num_seq and max_q_len thresholds
   - Replace tl.make_block_ptr with masked loads to avoid TMA alignment issues
   - Fix pages_uniform check to use kv_indptr instead of oversized kv_indices

3. Onboard nvidia/Gemma-4-31B-IT-NVFP4 model:
   - Add FP8 KV cache support to triton_paged attention: cast k/v from fp8
     to query dtype before tl.dot
   - Fix SDPA gather path to cast k_sdpa/v_sdpa when kv cache dtype differs
   - Fix *.embed_tokens exclusion in ModelOPTQuantConfigReader._ALWAYS_EXCLUDE
     so tied lm_head weights are not accidentally NVFP4-quantized

### CODE REVIEW

File: tensorrt_llm/_torch/auto_deploy/custom_ops/attention/triton_paged_attention.py [hunk 1]
```diff
@@ -938,11 +932,19 @@ def triton_paged_context(
 
     max_pages = (max_q_len + page_size - 1) // page_size
     total_expected_pages = num_seq * max_pages
+    # Force SDPA for large head_dim: the Triton paged kernel's tl.dot produces
```
Comment (bmarimuthu-nv) [inline tensorrt_llm/_torch/auto_deploy/custom_ops/attention/triton_paged_attention.py]: Can we add some units tests for this kernel update?

**Review by bmarimuthu-nv — APPROVED** [review]
LGTM, thanks!

**Review by venkywonka — APPROVED** [review]

**Comment by suyoggupta** [comment]
/bot run --extra-stage "DGX_B200-4_GPUs-AutoDeploy-1, DGX_H100-4_GPUs-AutoDeploy-1"

**Comment by suyoggupta** [comment]
/bot run --extra-stage "DGX_B200-4_GPUs-AutoDeploy-1, DGX_H100-4_GPUs-AutoDeploy-1"

**Comment by suyoggupta** [comment]
/bot run --extra-stage "DGX_B200-4_GPUs-AutoDeploy-1, DGX_H100-4_GPUs-AutoDeploy-1"

**Comment by suyoggupta** [comment]
/bot run --extra-stage "DGX_B200-4_GPUs-AutoDeploy-1, DGX_H100-4_GPUs-AutoDeploy-1"
